File: Super Reduced String.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'superReducedString' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string superReducedString(string s)
{
    int n = s.length();
    int top = 0;

    for (int i = 0; i < n; ++i)
    {
        if (top == 0 || s[top-1] != s[i])
        {
            s[top++] = s[i];
        }
        else
        {
            --top;
        }
    }

    return top == 0 ? "Empty String" : s.substr(0, top);
}


string superReducedString2(string s)
{
    int n = s.length();
    string result{};

    for (int i = 0; i < n; ++i)
    {
        if (result.empty() || result.back() != s[i])
        {
            result.push_back(s[i]);
        }
        else
        {
            result.pop_back();
        }
    }

    return result.empty() ? "Empty String" : result;
}


int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = superReducedString(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Super-Reduced-String.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'superReducedString' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static String superReducedString(String s) {
        StringBuilder sb = new StringBuilder();

        for (char c : s.toCharArray()) {
            if (sb.length() == 0 || sb.charAt(sb.length() - 1) != c) {
                sb.append(c);
            } else {
                sb.deleteCharAt(sb.length() - 1);
            }
        }

        return sb.length() == 0 ? "Empty String" : sb.toString();
    }

   public static String superReducedString2(String s) {
        char[] stack = s.toCharArray();
        int top = 0;

        for (int i = 0; i < stack.length; ++i) {
            if (top == 0 || stack[top - 1] != stack[i]) {
                stack[top++] = stack[i];
            } else {
                --top;
            }
        }

        return top == 0 ? "Empty String" : new String(stack, 0, top);
    }
}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = bufferedReader.readLine();

        String result = Result.superReducedString(s);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
